cmake/ccache.cmake

```
@@ -0,0 +1,192 @@
+#[[
+Copyright 2023 Antony Peacock
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+persons to whom the Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+]]
+
+include_guard(GLOBAL)
+
+
+#[=======================================================================[.rst:
+enable_ccacche
+------------------
+
+Overview
+^^^^^^^^
+
+Enable use of Ccache for compiler caching during the build process.
+
+.. code-block:: cmake
+
+  enable_ccacche(
+      [DEBUG]
+      [ICECC]<icecc path>
+      [LOGFILE]<ccache logfile path>
```

This is an optional parameter so does not take a logfile path.  
```suggestion
      [LOGFILE]
```

Two potential courses of action here, rename is `ENABLE_LOGFILE` so it's clearly an optional value, or change it to a single value so a path location can be configured dynamically.  I tend to prefer the first for simplicity.